src/compute/src/server.rs

```diff
@@ -838,3 +648,47 @@ impl<'w, A: Allocate + 'static> Worker<'w, A> {
         Ok(())
     }
 }
+
+/// Spawn a thread to bridge between [`ClusterClient`] and [`Worker`] channels.
+///
+/// The [`Worker`] expects a pair of persistent channels, with punctuation marking reconnects,
+/// while the [`ClusterClient`] provides a new pair of channels on each reconnect.
+fn spawn_channel_adapter(
```

Comment:
I considered making this a wrapper around `client_rx` instead to transform the command/response stream without spawning additional threads, similar to how the `GenericClients` work. But the thread version ends up much simpler to implement, so I stuck with that.

If we are worried with the overhead of having one additional thread per worker (no clue if we should be), we could make this a tokio task instead.

Reply:
Yeah, I wasn't sure if I could just call `mz_ore::task::spawn` or if I need to thread through a tokio runtime handle like we do elsewhere. But looks like `spawn` just works and it's also what's used in storage for the async worker.

Reply:
Ah, turns out it's not so simple. Using a task here means we have to change the command channel from the `ClusterClient` from a crossbeam channel to a `tokio::sync::mpsc` one, so we can do `recv().await` from it. But in the storage cluster that channel end is `recv`'d on in sync code where we can't `await`, so there we want a crossbeam channel instead.